### DISCUSSION TITLE
Installer: Option to do a Clean Install?

### DISCUSSION THREAD

**Discussion by gab3d** [opentoonz/opentoonz#6469] [Ideas]
We keep seeing the same problems (crashes) over and over again, when unaware people tries to upgrade OT to a newer version without previously uninstalling and removing/renaming the OpenToonz Stuff folder. 

So perhaps we should consider checking for previous installed versions, during the installation process, and if one is found, warn the user about it and offer to do a Clean Install by automatically uninstalling the old version and renaming the Stuff folder to something like "_old", so we can get rid of *most* this karma. 😸 

What do you think about it?

**Comment by DarrenTAnims** [comment]
100% yes.

If there are breaking changes to the settings, then:
A. Preferably the install should update the appropriate settings files, so the user gets to keep some of their settings (room layouts, keyboard shortcuts, etc)
B. or if the changes are too widespread, then the installer should rename the current OpenToonz stuff folder automatically before installing or tell the user to do it.

**Comment by artisteacher** [comment]
I completely agree.

**Comment by F-Burning** [comment]
I think I have never had a problem with this, but without a doubt it is a great option ✌️
